### PR TITLE
Runtime/Building Block/Servient Implementation Refinements

### PR DESCRIPTION
**Pull request by mmccool**
An attempt to address https://github.com/w3c/wot-architecture/issues/130 and resolve some ambiguities around whether or not the WoT Runtime is a required component, this PR includes revisions to chapter 6 and 7 that are based on the assumption that the WoT Runtime can be used WITHOUT the WoT Scripting API.  As such, it includes an update the definition of WoT Runtime in the Terminology section as well.

This PR also adds security content to the WoT Building Blocks and in a few other places.

### CODE REVIEW

**Comment by mmccool** [comment]
NOTE: based on the discussion in the main call, will either complete this PR (and layer a new abstraction on top of the WoT Runtime, making the WoT Runtime specfic and required for the Scripting API) OR create another PR and make the WoT Runtime required for a WoT Thing, but make the WoT Scripting API *optional* for the WoT Runtime.

File: index.html [hunk 1]
```diff
@@ -1762,79 +1769,103 @@ <h1>WoT Building Blocks</h1>
...
-			It is also possible, as shown in <a
-				href="#arch-thing-without-scripting"></a>, to use the WoT Thing
-			Description to described a device or service that does not use the
-			WoT Scripting API. There are two cases where this may be important:
-			pre-existing "brownfield" devices for which we are providing a Thing
-			Description in order to allow them to interoperate with other
-			elements of a WoT system; and devices which need to use a programming
-			language not supported by the WoT Scripting API.
+                <p>It is however possible to implement a WoT Thing <em>without</em> using the WoT Runtime
+                and WoT Scripting API as shown in <a href="#arch-thing-without-scripting"></a>.
```
Comment (zolkis) [inline index.html]: Somehow we need to make clear that the WoT Runtime is larger concept than the Scripting Runtime used in the implementation of the Scripting API.

File: index.html [hunk 2]
```diff
@@ -1910,17 +1941,18 @@ <h3>WoT Binding Templates</h3>
...
 				(Properties, Actions, and Events). It also allows scripts to specify
-				and implement Interaction Affordances and expose Things.</p>
+				and implement Interaction Affordances and expose Things (that is,
+                                define and publish a Thing Description).</p>
 			<p>
 				The <a>Scripting API</a> is an optional "convenience" building block
-				in WoT and it is typically used in network nodes that are able to
-				host a WoT Runtime, for instance gateways or more powerful IoT
-				devices.
+				in WoT.  It is used in combination with a WoT Runtime
```
Comment (zolkis) [inline index.html]: I wonder if we are making a constraint here. So far I understood that a WoT Runtime can exist without a scripting API implementation, just encapsulating the WoT network interface and the environment.

File: index.html (hunk 1 above)
Comment (mmccool) [reply index.html]: Yes, I think your comments were a little behind my edits.

**Comment by mmccool** [comment]
Ready to merge.  It only includes edits up to "Servient Interconnection", however.   But that should be OK since those sections are about external connections and this set of changes focuses on the internal architecture of Things and Servients.

**Comment by mlagally** [comment]
@mmccool Some initial review comments:
* The PR breaks the section structure of the chapters following chapter 7.
* Figure 24 should not contain the ThingDirectory, this will confuse the reader

**Comment by mlagally** [comment]
Agreed an arch call on 21.3.:

WoT Runtime: A runtime system that maintains an execution environment for applications and is able to expose and optionally consume a TD, offers a protocol binding to one or more protocols, and maintains private security metadata.

**Comment by mlagally** [comment]
Scripting Runtime:
A runtime system that is used for implementing an API in a scripting language (e.g. using ECMAScript, LUA, Python etc.)

**Pull request merged**